File: 231-PowerOfTwo/231-PowerOfTwo.cpp
// Last updated: 1/8/2026, 3:34:19 PM
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if(n<1) return false;
        while(n%2==0){
            n/=2;
        }
        return n==1;
    }
};

File: 231-PowerOfTwo/231-PowerOfTwo.java
// Last updated: 9/10/2026, 9:59:35 AM
class Solution {
    public boolean isPowerOfTwo(int n) {
        if(n<=0) return false;
        return (int)Math.pow(2,(int)(Math.log(n)/Math.log(2)))==n;
    }
}